### DISCUSSION TITLE
Barrel Roll mod affects gameplay cursor size

### DISCUSSION THREAD

**Discussion by Sinamer** [ppy/osu#15904] [Q&A] [answered]
When playing with Barrel Roll mod enabled, the cursor becomes smaller.
The option to adjust cursor size for beatmaps is off:
![image](https://user-images.githubusercontent.com/53346335/144445480-c01c82ad-b2c7-4bba-946b-1c7d0ba0072b.png)
![osu_2021-12-02_16-53-09](https://user-images.githubusercontent.com/53346335/144445834-173022d7-381c-44e0-9213-bf95bca7bfd4.png)
![osu_2021-12-02_16-53-00](https://user-images.githubusercontent.com/53346335/144445843-3e2bea60-d8fe-43f4-aa66-e2bd7882bd6b.png)

**Comment by bdach** [comment, marked as answer]
that would be because the entire playfield is shrunk down so that it can fit on the screen while it's rotating. so the effective circle size is also decreased. not sure what the expectation there is.

**Reply by Sinamer** [reply]
That's the point, the option to adjust the cursor size is turned **off**, because I don't want the cursor to change its size. It is supposed to be constant. If I wanted for cursor to change the size, then I would turn the option on. Right now, the cursor size changes regardless of the option, when you enable Barrel Roll mod.

**Reply by bdach** [reply]
but what that option does, is scale the cursor with accordance to a given *beatmap's* circle size. this is a different scenario in that the *mod itself* is scaling the *entire playfield* down, objects and the cursor included.

i suppose the cursor could be enlarged back up by the same constant to compensate, although i don't know if that matches expectation given that it will increase the effective cursor size relative to the objects' size.

**Reply by Sinamer** [reply]
Yes, I do understand that it's the intended behaviour of the mod to scale the entire playfield down, and that the option description doesn't mention any adjustments made by mods. But it feels wrong to scale the cursor just because you have a mod enabled.

From a user perspective, launching a beatmap, and the cursor size being different, even though all settings that seem related are turned off, is confusing. I, for one, noticed it was wrong immediately.
If it is more than intended, consider this discussion just a suggestion to make cursor size constant, and repurpose the option in settings to disallow any cursor adjustment game-wide unless the user says so explicitly.

Edit: Now that I think about it, such a limitation would hurt the ability to create mods that mess with cursor size (like, making it variable), that didn't cross my mind in the slightest. I agree then, it should be up to the mod implementation to use the cursor size.
Sorry for the noise.

**Reply by peppy** [reply]
Not really valid reasoning for this occurring, but it's not a simple one to fix. We'll see how many people ask for this change and react accordingly.
